query_sheet.rb

```diff
@@ -186,5 +208,126 @@ def add_errors_to_message(
   error_message += ad_message + "\n"
 end
 
+def candidates_by_attribute(candidates, attribute, one_to_many=false)
+  candidates.select do |candidate|
+    candidate[attribute] &&
+      candidate[attribute].length > 0
+  end.reduce({}) do |obj, candidate|
+    if obj[candidate[attribute]]
+      obj[candidate[attribute]] << candidate
+    else
+      obj[candidate[attribute]] = one_to_many ? [candidate] : candidate
+    end
+    obj
+  end
+end
+
+def find_diffs(imported_candidates)
```

Comment:
May as well remove this, my branch has something very similar.

Reply:
#deprecation